.github/workflows/build-project.yaml

```
@@ -297,8 +297,8 @@ jobs:
           build-bundle: ${{ fromJSON(needs.check-event.outputs.package) }}
           bundle: obs-studio-flatpak-${{ needs.check-event.outputs.commitHash }}.flatpak
           manifest-path: ${{ github.workspace }}/build-aux/com.obsproject.Studio.json
-          cache: ${{ fromJSON(steps.setup.outputs.cacheHit) || (github.event_name == 'push' && github.ref_name == 'master')}}
-          restore-cache: ${{ fromJSON(steps.setup.outputs.cacheHit) }}
+          cache: ${{ fromJSON(steps.setup.outputs.cacheHit) || ((github.event_name == 'push' || github.event_name == 'scheduled') && github.ref_name == 'master')}}
+          restore-cache: ${{ fromJSON(steps.setup.outputs.cacheHit) && github.event_name != 'scheduled' }}
```

```suggestion
          restore-cache: ${{ fromJSON(steps.setup.outputs.cacheHit) }}
```

This will prevent `scheduled` jobs to restore cache if a matching one is already created, which we only re-create when the manifest changes.

Settings up `cache` to work `push` or `schedule` (only on master) should be enough.

---

IIRC, we do not want the scheduled job to use the cache. We only want the scheduled job to create caches.

---

Then the action will likely log an error about a already existing cache with the same key.

Flatpak caching is only managed by Flatpak related GHA code, the removal of stale is only for Windows, macOS and Ubuntu `ccache` cache.

---

Yes, that is currently true. If we adjust this, I will probably add Flatpak cache removal to the scheduled workflow.